ln-gateway/src/lib.rs

```diff
@@ -282,17 +288,16 @@ impl LnGateway {
     }
 
     async fn handle_htlc_incoming_msg(&self, htlc_accepted: HtlcAccepted) -> Result<Preimage> {
-        let amount = htlc_accepted.htlc.amount;
+        let invoice_amount = htlc_accepted.htlc.amount;
         let payment_hash = htlc_accepted.htlc.payment_hash;
-        let rng = rand::rngs::OsRng::new().unwrap();
-
-        tracing::debug!("Incoming htlc for payment hash {}", payment_hash);
-        let (outpoint, _) = self.buy_preimage_offer(&payment_hash, &amount, rng).await?;
-        tracing::debug!("Decrypting preimage {}", payment_hash);
-        let preimage = self.await_preimage_decryption(outpoint).await?;
-        tracing::debug!("Decrypted preimage {:?}", payment_hash);
+        let mut rng = rand::rngs::OsRng::new().unwrap();
```

Comment:
`unwrap` could be `expect("only systems with a randomness source are supported")` (or just use `thread_rng`?)